### ISSUE TITLE
Unexpected resolution of scoped instances (when using `declare`)

### ISSUE THREAD

**Issue by jcornaz** [InsertKoinIO/koin#1191]
**Describe the bug**

Please consider the following code:

```kotlin
class MyScope

class MyScopedData(var name: String)

private val module = module {
    scope<MyScope> {
        scoped { MyScopedData("") }
    }
}

fun main() {
    val app = startKoin {
        modules(module)
    }

    // Create two scopes
    val scopeA = app.koin.createScope<MyScope>(UUID.randomUUID().toString(), MyScope())
    val scopeB = app.koin.createScope<MyScope>(UUID.randomUUID().toString(), MyScope())

    // Declare scope data on each scope
    scopeA.declare(MyScopedData("A"))
    scopeB.declare(MyScopedData("B"))
    
    // Get scope of each data
    println(scopeA.get<MyScopedData>().name) // <-- I expect to get the scope data declared on ScopeA
    println(scopeB.get<MyScopedData>().name) // <-- I expect to get the scope data declared on ScopeB
}
```

If we run this code, we get: 

```
B
B
```

And we would expect to see:

```
A
B
```

Koin version: 3.1.2

**Comment by Khaled431** [comment]
I am also getting this bug.

**Comment by arnaudgiuliani** [comment]
Fixed in 3.2.0
